src/Middleware/RateLimiting/src/EnableRateLimitingAttribute.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Metadata that provides endpoint-specific request rate limiting.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+public sealed class EnableRateLimitingAttribute : Attribute
+{
+    /// <summary>
+    /// Creates a new instance of <see cref="EnableRateLimitingAttribute"/> using the specified policy.
+    /// </summary>
+    /// <param name="policyName">The name of the policy which needs to be applied.</param>
+    public EnableRateLimitingAttribute(string policyName)
+    {
+        ArgumentNullException.ThrowIfNull(policyName);
+
+        PolicyName = policyName;
+    }
+
+    internal EnableRateLimitingAttribute(DefaultRateLimiterPolicy policy)
+    {
+        Policy = policy;
+    }
+
+    /// <summary>
+    /// The name of the policy which needs to be applied.
+    /// </summary>
+    public string? PolicyName { get; }
```

Comment:
This feels super weird, from a consumer perspective there is no way this can be null.

Reply:
Sure it can. Add an inline policy using the `RequireRateLimiting` extension method and then look at the endpoint metadata. I get the point about not being able to trivially construct such an attribute, but hopefully we can add more first-class support for inline policies later.